spotify_project/albums/tests/test_utils.py

```diff
@@ -0,0 +1,66 @@
+from operator import itemgetter
+
+from albums.factories import AlbumFactory
+from albums.models import Album
+from albums.utils import AlbumUtils
+from artists.factories import ArtistFactory
+from testing import BaseTestCase
+
+
+class AlbumUtilsTestCase(BaseTestCase):
+    def setUp(self):
+        self.album_utils = AlbumUtils()
+        self.artists = ArtistFactory.create_batch(3)
+        self.albums = AlbumFactory.create_batch(3, artists=self.artists)
+        # not included
+        AlbumFactory(artists=[ArtistFactory()])
+
+    def test_add_albums_to_db(self):
+        with self.assert_num_objects_created({Album: 3}):
+            self.album_utils.add_albums_to_db(self.dummy_library_data)
+
+    def test_get_all_albums(self):
+        actual_albums = list(self.album_utils.get_all_albums(self.dummy_library_data))
+        expected_albums = [track['album'] for track in self.dummy_library_data]
+        for count, album in enumerate(actual_albums):
+            self.assertEqual(album.pk, expected_albums[count]['id'])
+            self.assertEqual(album.name, expected_albums[count]['name'])
+
+    def test_get_new_albums(self):
+        unsaved_albums = AlbumFactory.build_batch(2)
```

Comment:
This is likely why you ran into issues with the `SubFactory`. This line was probably yelling at you about FK constraints. You can probably get around that by building the artist objects first, then passing them in here. That will likely let you get back to use `SubFactory`.